server/http/middleware/tenant.go

```diff
@@ -0,0 +1,44 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/google/uuid"
+)
+
+type key string
+
+var (
+	TenantIDKey key = "tenantID"
+)
+
+const HeaderTenantID = "X-Tracetest-TenantID"
+
+func TenantMiddleware(next http.Handler) http.Handler {
```

Comment:
Validates that the tenant id has the proper format, if all is correct, we add it to the context so the rest of the app can use it